### PR TITLE
Fix crash panic in operator when policy has empty node selector

### PR DESCRIPTION
**Pull request by SchSeba**
The commit will fix a nil pointer exception in the operator when there is a policy with an empty selector the update of the devicePlugin daemonSet will fail

### CODE REVIEW

**Comment by SchSeba** [comment]
@zeeke @Eoghan1232 @e0ne please take a look for this quick fix

File: controllers/helper.go [hunk 1]
```diff
@@ -302,17 +365,7 @@ func syncDaemonSet(ctx context.Context, client k8sclient.Client, scheme *runtime
 
 		if equality.Semantic.DeepEqual(in.OwnerReferences, ds.OwnerReferences) &&
 			equality.Semantic.DeepDerivative(in.Spec, ds.Spec) {
-			// DeepDerivative has issue detecting nodeAffinity change
```
Comment (mlguerrero12) [inline controllers/helper.go:305]: Why is this part not needed anymore?

File: controllers/helper.go [hunk 2]
```diff
@@ -302,17 +365,7 @@ func syncDaemonSet(ctx context.Context, client k8sclient.Client, scheme *runtime
...
-			// https://bugzilla.redhat.com/show_bug.cgi?id=1914066
-			// DeepDerivative doesn't detect changes in containers args section
-			// This code should be fixed both with NodeAffinity comparation
-			if equality.Semantic.DeepEqual(in.Spec.Template.Spec.Affinity.NodeAffinity,
-				ds.Spec.Template.Spec.Affinity.NodeAffinity) &&
-				equality.Semantic.DeepEqual(in.Spec.Template.Spec.Containers[0].Args,
-					ds.Spec.Template.Spec.Containers[0].Args) {
-				logger.V(1).Info("Daemonset spec did not change, not updating")
```
Comment (mlguerrero12) [inline controllers/helper.go:313]: I think the log is still valid before `return nil`

File: controllers/helper.go (hunk 1 above)
Comment (SchSeba) [reply controllers/helper.go:305]: because `equality.Semantic.DeepDerivative` is able to handle that

File: controllers/helper.go (hunk 2 above)
Comment (SchSeba) [reply controllers/helper.go:313]: I didn't remove I just move the functions to the right file.

before this change functions in helpers called functions in the controller file that breaks the point of helpers

Comment (mlguerrero12) [reply controllers/helper.go:313]: I meant this line

`logger.V(1).Info("Daemonset spec did not change, not updating")`

Comment (SchSeba) [reply controllers/helper.go:313]: ohhhh you are right! sorry stupid me :(

**Comment by SchSeba** [comment]
@mlguerrero12 done thanks! :)

**Review by mlguerrero12 — APPROVED** [review]
LGTM

**Comment by SchSeba** [comment]
Hi @zeeke @e0ne @Eoghan1232 please take a look when you have some time this is a small fix

**Review by zeeke — APPROVED** [review]
LGTM

**Comment by SchSeba** [comment]
Thanks folks!
merging

**Pull request merged**